### PR TITLE
chore: improve pull request workflows

### PR DESCRIPTION
**Pull request by juliapampus**
## What this PR changes/adds

Add new regex that is more restrictive. PR titles have to match the conventional commit prefixes. Add label checker.

## Why it does that

Improve PR quality. Label check was suggested in #1728.

## Further notes

...

## Linked Issue(s)

Relates #1719

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: .github/workflows/scan-pull-request.yaml [hunk 1]
```diff
@@ -34,11 +34,23 @@ jobs:
...
+          prefix_case_sensitive: true
+
+  check-for-assigned-labels:
+    runs-on: ubuntu-latest
+    continue-on-error: false
+    steps:
+      - uses: actions/checkout@v3
+      - uses: agilepathway/label-checker@v1.0.113
+        with:
+          any_of: api,bug,build,dependencies,documentation,enhancement,ids,java,no-changelog,policy,refactoring
```
Comment (ndr-brt) [inline .github/workflows/scan-pull-request.yaml:55]: I would propose to choose the labels based on which categories we want to see in the changelog.
e.g. currently now there are only Bugfixes (bug), Documentation (documentation) and New Features & Improvements (*).
I would keep `bug`, `documentation`, `enhancement` (that could be used for `new features`), `no-changelog`. 
maybe also `api`, `dependencies` and `refactoring` (that could be added as categories in the `changelog` task configuration)
but I would avoid the others (`build`, `ids`, `java`, `policy`).

Makes sense?

Comment (juliapampus) [reply .github/workflows/scan-pull-request.yaml:55]: yes.. the problem is that there is no action that allows wildcards, so e.g. changes to the pipeline would need the `build` label as no PRs with 0 labels are allowed. Maybe just skip `ids`, `java`, `policy` as this is too specific? The other ones kind of match the conventional commits prefixes.

Comment (juliapampus) [reply .github/workflows/scan-pull-request.yaml]: I would suggest to take a look at adding further categories to the changelog later.. first step is to have labels.

**Review by ndr-brt — APPROVED** [review]
good for me

**Pull request merged**